### PR TITLE
Complete exercies of chapter/8

### PR DESCRIPTION
**Pull request by nguyentrieutien1**
+ attr_accessor  used to create getter and setter methods for a class 
+ Instance Method related to a specific instance of a class, can access and manipulate the data of an instance
+ Class Method related to the class rather than a specific instance, dan operate on class-level data rather than instance-specific data.
+ Cookies are small pieces of data stored in a user's browser, often used to track login status, user information
+  update_attribute used to update a single attribute of an object, opposite update_attributes update_columns but update_columns can update multiple columns at one.
+ update_attribute! same update_attribute but if update fail, this raise error
+  update_column updates a specific column in the database without running callbacks or validations, same 
+ Save used to store data an object in the db, return true if the save is successful, false otherwise.
+ Save! same save, but raise error if save fail.
+ self in Rails used refer to the current object ( based on context  )

*** Content needed note in chapter/8

+ Using authenticate method to valid the password requested to the password in db
+  If user login -> handle reset the session -> call method in users helper to store the session that users
+ Check user is logined in users helper
+ For log out, Using button instead of link_to because  link_to using GET method ( default ) and destroy use DELETE method
+ Install jquery-rails to use dropdown in bootstrap and more... 
+ Remember include SessionsHelper to use these module

### CODE REVIEW

**Comment by nguyentrieutien1** [comment]
image

**Comment by nguyentrieutien1** [comment]
ready
image

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class SessionsController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.find_by name: params[:name]&.downcase
```
Comment (vanvtt-0952) [inline app/controllers/sessions_controller.rb]: KHông sai nhưng chị có suggest để code clear hơn

em dùng action filter (cụ thể là before_action) để call method load_user
còn method load_user sẽ handle việc tìm user theo name và dồng thời handle nếu user nil luôn

còn methof create ở đây em chỉ handle authenticate và cũng ko cần toán từ an toàn &. `@user&.authenticat` lun nhá

**Comment by nguyentrieutien1** [comment]
ready

**Comment by manhhnee** [comment]
Merge

**Pull request merged**